### DISCUSSION TITLE
Why is “Checkbox” rendered as a button, not an input? What’s the rationale behind that choice?

### DISCUSSION THREAD

**Discussion by getflourish** [radix-ui/primitives#2878] [Help]
Honest question: 

I would like to understand why the Checkbox and Switch components have been implemented as button elements with ARIA attributes. 

What are the advantages over using a native input type="checkbox"?

**Comment by ghost** [comment]
Have you found out the reason ?
Curious to know about it as well

**Reply by enix79** [reply]
https://github.com/radix-ui/primitives/discussions/874 - have you seen this discussion?

**Reply by getflourish** [reply]
Yes seen it, but that doesn't provide a rationale.
